tests/find/test-suite-1/test.default-index.js

```diff
@@ -390,4 +390,90 @@ testCases.push(function (dbType, context) {
           });
       });
   });
+
+  it('null values are indexed', function () {
+      var db = context.db;
+      return db.createIndex({
+          index: {
+              fields: ['foo']
+          }
+      }).then(function () {
+          return db.bulkDocs([
+              {_id: '1', foo: null},
+              {_id: '2', foo: 1},
+              {_id: '3', foo: 2},
+              {_id: '4', foo: 3}
+          ]).then(function () {
+              return db.find({
+                  selector: {
+                      foo: {$ne: 1}
+                  },
+                  fields: ['_id']
+              }).then(function (resp) {
+                  resp.docs.should.deep.equal([
+                    {_id: "1"},
+                    {_id: "3"},
+                    {_id: "4"},
+                  ]);
+              });
+          });
+      });
+  });
+
+  it('handles booleans as a valid index value', function () {
+      var db = context.db;
+      return db.createIndex({
+          index: {
+              fields: ['foo']
+          }
+      }).then(function () {
+          return db.bulkDocs([
+              {_id: '1', foo: true},
+              {_id: '2', foo: false},
+              {_id: '3', foo: 2},
+              {_id: '4', foo: 3}
+          ]).then(function () {
+              return db.find({
+                  selector: {
+                      foo: {$eq: true}
+                  },
+                  fields: ['_id']
+              }).then(function (resp) {
+                  resp.docs.should.deep.equal([{_id: '1'}]);
+              });
+          });
+      });
+  });
+
+  it.skip('boolean values are indexed', function () {
+      var db = context.db;
+      return db.createIndex({
+          index: {
+              fields: ['foo']
+          }
+      }).then(function () {
+          return db.bulkDocs([
+              {_id: '1', foo: true},
+              {_id: '2', foo: false},
+              {_id: '3', foo: 2},
+              {_id: '4', foo: 3}
+          ]).then(function () {
+              return db.find({
+                  selector: {
+                      foo: {$ne: 2}
```

Comment:
A `$ne` should not use an index. So you can remove the createIndex above or you need another selector on `foo` to force it to use the `foo` selector. This could be why PouchDB and CouchDB are different here. Its worth running an explain and seeing what PouchDB and CouchDB are doing here. But PouchDB could be using the index.